### PR TITLE
CHI-2566 Teams view skills filtering proposal

### PR DESCRIPTION
**Pull request by mythilytm**
### Checklist
- [ ] Corresponding issue has been opened
- [ ] New tests added
- [ ] Feature flags added
- [ ] Strings are localized
- [ ] Tested for chat contacts
- [ ] Tested for call contacts

### Other Related Issues

None

### AFTER YOU MERGE

1. Cut a release tag using the Github workflow. Wait for it to complete and notify in the #aselo-deploys Slack channel.
2. Comment on the ticket with the release tag version AND any additional instructions required to configure an environment to test the changes.
3. Only then move the ticket into the QA column in JIRA

You are responsible for ensuring the above steps are completed. If you move a ticket into QA without advising what version to test, the QA team will assume the latest tag has the changes. If it does not, the following confusion is on you! :-P

### CODE REVIEW

File: plugin-hrm-form/src/components/teamsView/Filters.ts [hunk 1]
```diff
@@ -38,26 +38,38 @@ const activityNoOfflineByDefault: FilterDefinitionFactory = (appState, _teamFilt
   };
 };
 
-/**
- * This function returns a list of skills defined in the taskrouter_skills configuration
- */
-const skillsFilterDefinition: FilterDefinitionFactory = (_appState, _teamFiltersPanelProps) => {
-  const skillsArray = Manager.getInstance().serviceConfiguration.taskrouter_skills?.map(skill => ({
+const skillsArray = Manager.getInstance().serviceConfiguration.taskrouter_skills?.map(skill => {
```
Comment (nickhurlburt) [inline plugin-hrm-form/src/components/teamsView/Filters.ts:41]: I could be wrong, but I would guess that if `skillsArray` is defined here, then this code will only get executed once when Flex starts up. Thus if someone adds or removes skills via the Flex Admin page, someone will have to reload Flex to get them. (Again I could be wrong -- you could do a quick test to see).

To allow skills to update more frequently and also have DRY code, you could make this a function, eg.
```
const getSkillsArray = () => {
 return Manager.getInstance().serviceConfiguration.taskrouter_skills?.map(skill => {
    value: skill.name,
    label: skill.name,
  });
}
```
and then call that function in the `FilterDefinitionFactory`s below.

**Review by nickhurlburt — APPROVED** [review]
LGTM given one comment below, and pending QA acceptance

File: plugin-hrm-form/src/components/teamsView/Filters.ts (hunk 1 above)
Comment (mythilytm) [reply plugin-hrm-form/src/components/teamsView/Filters.ts:41]: you are right in suspecting this

**Comment by nickhurlburt** [comment]
@mythilytm I'm not sure your most recent change here actually changed it to a function that will be executed every time... I think it's still just executing once on startup. Unless you've tested this and it worked as intended?

In addition to the change in my previous comment (`const getSkillsArray = () =>` etc), you'd need to change the factory functions to be more like:
```
  const skillsArray = getSkillsArray();
  return {
    id: 'data.attributes.routing.skills',
    title: 'Enabled Skills',
    fieldName: 'skills',
    type: FiltersListItemType.multiValue,
    options: skillsArray ? sortBy(skillsArray, ['label']) : [],
    condition: 'IN',
  };
```

**Pull request merged**